### DISCUSSION TITLE
Sampleset encryption

### DISCUSSION THREAD

**Discussion by Psi58** [GrandOrgue/grandorgue#1883] [General]
This might be heretical, but I've found myself wondering if the option of sampleset encryption would help secure and advance the future of GO.

Many organ samplesets are commercial products, and unavailable to GO users because there's no way for providers to charge for them, other than relying on users voluntarily making a contribution.

This means that (a) GO users lack access to some fantastic samplesets and (b) commercial sampleset producers have no viable way to introduce their products to GO users.

If there was a way to add **_optional_** encryption to samplesets (this is a big assumption - others with more technical know-how than me can advise whether this is feasible), the producers could get access to a larger market, and users access to a greater range of organs - which seems like a "win-win". And if the result is greater usage of GO then the whole project would benefit.

Of course the encryption functionality should only be optional, so the free samplesets that generous-spirited enthusiasts make available will remain unaffected, and GO will remain a bona fide open source project.

Presumably (if the idea isn't shot down) the next step would be to verify with a few sampleset producers that they would support the idea in principle.

But first, what do other users think, both in principle and also whether it's achievable in practice?

**Comment by eturpault** [comment]
Personnaly I think that with the possibility that we have now to use non encrypted Hauptwerk sample sets in GrandOrgue (thanks to OdfEdit), GrandOrgue users have an additional large choice of accessible quality sample sets (free and commercial) to satisfy their aspirations. For example Sonus Paradisi proposes 115 non encrypted sample sets (and 30 encrypted), and in Piotr Grabowski only the latest published sample set (Düsseldorf, St. Lambertus) is encrypted.
In addition, I am doubting that sample set producers would be interrested in making GrandOrgue versions of their Hauptwerk sample sets, as it is the case of Piotr Grabowski as explained here (https://piotrgrabowski.pl/info/) in answer to the question "Will there be GrandOrgue versions of newer sample sets?".
Last, management of protected data by an open source software should be quite easy to workaround.
Let's see what other users think.

**Comment by midimusic** [comment]
1. GO sample sets already take  long enough to load. Encrypted sample sets will automatically take longer because they have to be decrypted when loading.
2. There are already sufficient free sample sets available for GO so why would anybody want to pay for more?
3. If you really want to pay for using your organ you can always choose to use HW.

csw900

**Comment by Bumblebee001** [comment]
Sadly, man is what he is...... greedy. Ultimately it is this greed that leads to destruction. 

With commercialisation, greed will inevitably follow. Watch what has been happening with HW. I will not go into the details. Also look at the prices of several sample-sets...... ludicrous! People are not happy charging $100..... $1000 is better and more exciting.

I would love to earn a lucrative income from my sample-sets..... who wouldn't? I could do with a lot of things I currently cannot afford. However, I chose not to... and I don't expect others to do the same. It's up to each individual.

Progress however has been thanks to the generous champions who have given up lustful greed for the benefit of others. Such has been the case with GO, GOedit and GOodf and may other projects starting from Internet itself! Scores of people around the world would not have been able to venture into this field and enjoy playing wonderful organs of such good quality had it not been people like me/us who made the software and sample-sets freely available - at no charge at all. Just look at all the software and applications you use on a daily basis and are now almost dependent on and how much of it is free! 

Would we be sitting here discussing this topic if were otherwise?

**Reply by thbeckpr** [reply]
What sample sets have you produced and where can I find them?

**Reply by Bumblebee001** [reply]
http://mhpo.epizy.com

**Comment by MStraeten** [comment]
is there really an important amount of users willing to pay for encrypted go organ samplesets but not for the commercial alternative organ software?
So it seems to be a lot of effort for sample set makers to support two different encryption for just a handful users …

**Comment by Psi58** [comment]
Many thanks for the thoughtful comments about this proposal. It seems that the consensus is sceptical at best, or rather negative, so I'll let it drop. 

As I see it:

1) The assumption that many users are keen to access more samplesets is questionable. I had formed this view  because soon after I started my GO learning experience (about a year ago) I saw quite a few posts from users wishing they had access to certain HW samplesets, eg for English cathedrals. Of course, as the saying goes "the grass is always greener on the other side of the fence", and the actual demand may be not so significant. Also the excellent, and growing, capabilities of ODFedit (thank you @eturpault!) may have satisfied some of that demand.

2) You have highlighted specific technical issues that could make the proposal a non-starter - (a) the difficulty in creating secure, open-source encryption, (b) the impact on load time

3) There is understandable enthusiasm from the community (myself included) for free-of-charge samplesets. However, commercial producers exist, and will continue to do so. If their work is more than a hobby then covering their costs of time, equipment, access to church buildings etc seems not unreasonable. 
But those of us who cannot afford a HW license (or simply want to stay loyal to GO) have no way of benefiting from their work. (Slightly related to this, [I asked](https://github.com/GrandOrgue/grandorgue/discussions/1746) a little while ago how those of us who cannot afford HW might contribute to GO development).

Thank you again for your responses, I hope others also found it useful. I'll probably close the discussion in 1 week, unless there are more comments that should be aired in the meantime.

**Reply by MStraeten** [reply]
existing encrypted samplesets can't be supported since Hauptwerk won't share their encryption algorithms. 
So it requires sample set producers to support not just two different organ software with different odf standards but also two different encryption standards: hauptwerk and an - maybe open source - alternative. 
Doubling the effort to provide sample sets without a reasonable increase of sales and revenue is quite more relevant than technical difficulties ...
